lib/arg_error.rb

```diff
@@ -0,0 +1,17 @@
+module ArgError
```

Comment:
This is very very cool! Did you write this yourself or find it on the internet somewhere? It's totally cool if you found it on the internet! But we do ask that you site your source if it was sourced from somewhere.

Reply:
Hi Jared! I wrote this module myself. I learned module callbacks from some website, but I figured I was okay not needing to cite sources since I was writing all my own code and just using the concepts from he site.

Reply:
That is all absolutely true! So cool! It just looked so generic I thought there was a good chance you found that exact code somewhere. But even more amazing that you wrote it yourself. 
This looks to me like a module you can continue making use of in future projects!